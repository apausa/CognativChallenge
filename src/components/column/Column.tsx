import React from "react";
import { Text, View, TouchableOpacity, Image } from "react-native";
import styles from "../../screens/home/homeStyles";
import { Recipe } from "../../types/interface";

const Row: React.FC<any> = (item: Recipe) => {
  const PLACEHOLDER = "http://via.placeholder.com/640x360";
  const imageUrl = (item: Recipe) => {return (item && item.photo) ? item.photo : PLACEHOLDER}
  const renderImage = (item: Recipe) => (
    <Image source={{ uri: imageUrl(item) }} style={styles.recipeImage} />);
  return (
    <TouchableOpacity>
      <View style={styles.rowContainer}>
        <View style={styles.imageContainer}>{renderImage(item)}</View>
        <View style={styles.infoContainer}>
          <Text style={styles.categoryRow}>{item.categoryName}</Text>
          <Text style={styles.titleRow}>{item.name}</Text>
          <View style={styles.properties}>
            <View style={styles.cellRow}>
              <Text style={styles.cellText}>{item.duration} minutes</Text>
            </View>
            <View style={styles.cellRow}>
              <Text style={styles.cellText}>{item.complexity}</Text>
            </View>
            <View style={styles.cellRow}>
              <Text style={styles.cellText}>{item.people} people</Text>
            </View>
          </View>
        </View>
      </View>
    </TouchableOpacity>
  )
}

export default Row;
