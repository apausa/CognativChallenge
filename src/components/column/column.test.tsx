import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import Column from './Column';
import itemMock from '../../mocks/itemMock';
import navigationMock from '../../mocks/navigationMock';

describe('Given a Column component', () => {
  let screen: any;
  beforeEach(() => { screen =
    render(<Column item={itemMock} navigation={navigationMock}/>);
  })
  describe('When is rendered', () => {
    test('Then matches snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
    describe('And you press an item', () => {
      test('Then calls "navigation.push()"', () => {
        const item = screen.getByTestId('itemColumn');
        fireEvent.press(item);
        expect(navigationMock.push).toHaveBeenCalled();
      })
    })
  })
})
