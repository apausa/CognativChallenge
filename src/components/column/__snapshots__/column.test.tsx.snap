// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Column component When is rendered Then matches snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "#ffffff",
        "borderRadius": 4,
        "justifyContent": "flex-start",
        "marginBottom": 16,
        "marginHorizontal": 16,
        "opacity": 1,
        "overflow": "hidden",
      }
    }
    testID="itemColumn"
  >
    <View
      style={
        Object {
          "height": 128,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "http://via.placeholder.com/640x360",
          }
        }
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "justifyContent": "center",
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9e9e9e",
            "fontSize": 10,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "letterSpacing": 1,
            "lineHeight": 24,
            "textAlign": "left",
          }
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#333333",
            "fontSize": 16,
            "fontStyle": "normal",
            "fontWeight": "500",
            "letterSpacing": 0.5,
            "lineHeight": 24,
            "textAlign": "left",
          }
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginRight": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#bdbdbd",
                "fontSize": 13,
                "fontStyle": "normal",
                "fontWeight": "300",
                "letterSpacing": 0.5,
                "lineHeight": 24,
              }
            }
          >
            0
             minutes
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginRight": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#bdbdbd",
                "fontSize": 13,
                "fontStyle": "normal",
                "fontWeight": "300",
                "letterSpacing": 0.5,
                "lineHeight": 24,
              }
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginRight": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#bdbdbd",
                "fontSize": 13,
                "fontStyle": "normal",
                "fontWeight": "300",
                "letterSpacing": 0.5,
                "lineHeight": 24,
              }
            }
          >
            0
             people
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
